core.go

```diff
@@ -6,49 +6,53 @@ import (
 	"runtime"
 )
 
-var functionMap map[string]interface{}
+var functionMap map[string][]interface{}
 
 func init() {
-	functionMap = make(map[string]interface{})
+	functionMap = make(map[string][]interface{})
 	runtime.GOMAXPROCS(runtime.NumCPU())
 }
 
-func add(event string, task interface{}) error {
-	if _, ok := functionMap[event]; ok {
-		return errors.New("Event Already Defined")
-	}
-	functionMap[event] = task;
-	return nil
+func add(event string, task interface{}) {
+	functionMap[event] = append(functionMap[event], task);
 }
 
-
-func invoke(event string, params ...interface{}) ([]reflect.Value, error) {
-	f := reflect.ValueOf(functionMap[event])
-	if len(params) != f.Type().NumIn() {
-		return nil, errors.New("Parameter Mismatched")
-	}
-	in := make([]reflect.Value, len(params))
-	for k, param := range params {
-		in[k] = reflect.ValueOf(param)
+func invoke(event string, params ...interface{}) ([][]reflect.Value, error) {
+	result := make([][]reflect.Value, 0)
+	for _, regif := range(functionMap[event]) {
+		f := reflect.ValueOf(regif)
+		if len(params) != f.Type().NumIn() {
+			return nil, errors.New("Parameter Mismatched")
+		}
+		in := make([]reflect.Value, len(params))
+		for k, param := range params {
+			in[k] = reflect.ValueOf(param)
+		}
```

Comment:
@faide I don't think this will work in real time for the scenario

if a register 3 functions with different number of parameters like
```go
trigger.On("e-1", func(a, b int) { })
trigger.On("e-1", func(b string) { })
trigger.On("e-1", func() {} )
trigger.On("e-1", func(c string, d int) {})

// and now call
trigger.Fire("e-1", 1, 2, "a", "b", 10)
```
This will fail at line 24. we need to determine which parameters belongs to whom and send those to it. the remaining others for the next. 

we can do this by serializing parameters. the first n provided params for the first handler function with n parameters. or we can introduce a param type for the params.  in a basic like this
```go
type Param struct {
    params []interface{}
}
```
and change the `invoke(event string, params ...interface{})` to 
like this `invoke(event string, params ...Param)`. this could work fine with others.
But to initiate a Param we need an simpler system.